### PR TITLE
[NEW] Open uikit contextual bar

### PR DESCRIPTION
**Pull request by thassiov**
# What? :boat:

# Why? :thinking:

# Links :earth_americas:

# PS :eyes:

### CODE REVIEW

File: src/definition/uikit/UIKitInteractionPayloadFormatter.ts [hunk 1]
```diff
@@ -26,6 +26,31 @@ export function formatModalInteraction(view: IUIKitModalViewParam, context: IUIK
     };
 }
 
+export function formatContextualBarInteraction(view: IUIKitContextualBarViewParam, context: IUIKitInteraction): IUIKitContextualBarInteraction {
+    if (![UIKitInteractionType.CONTEXTUAL_BAR_OPEN,
+        UIKitInteractionType.CONTEXTUAL_BAR_UPDATE,
+        UIKitInteractionType.CONTEXTUAL_BAR_CLOSE].includes(context.type)) {
```
Comment (d-gubert) [inline src/definition/uikit/UIKitInteractionPayloadFormatter.ts:32]: If we use a type guard function to validated `context.type`, Typescript will be able to tell the actual type of the property

**Review by d-gubert — CHANGES_REQUESTED** [review]

**Comment by cuonghuunguyen** [comment]
as we had a permission system with Apps Engine, should we allow apps to display Contextual Bar and  Modal without the initial interaction from users?
I am making an app for livechat agents to send transfer request to the other agents. Without modals I am having to use a very compilated custom script

**Comment by d-gubert** [comment]
> as we had a permission system with Apps Engine, should we allow apps to display Contextual Bar and Modal without the initial interaction from users?

We were not considering making that change, as it is somewhat risky... as in, without this restriction, what will stop apps from spamming modals to users? 🤔

**Comment by graywolf336** [comment]
> We were not considering making that change, as it is somewhat risky... as in, without this restriction, what will stop apps from spamming modals to users? 🤔
The developers themselves and the marketplace publisher review team. What's to stop an app from spamming messages to users? If we use this same logic in other areas then we willcontinuouslyy hinder apps and what they can do. In my opinion

**Comment by d-gubert** [comment]
> The developers themselves and the marketplace publisher review team.

Having to check for this kind of "bad" behavior in the review process is not scalable.

We've decided to follow this practice as it is something Slack also does, and it looked very reasonable. But we can discuss about this again, of course.


> What's to stop an app from spamming messages to users?

This argument could go the other way as well. Like, shouldn't we implement some kind of "rate limiter" to say the least so apps would not abuse this power? Might fall in the same discussion mentioned above 🤷

**Comment by graywolf336** [comment]
In all honesty, I think us erroring on the "safe side" is curtailing creativity more than it's preventing people from spamming. Just my two cents :)

**Comment by cuonghuunguyen** [comment]
Agree with @graywolf336, we can't prevent all the cases of spamming. But imagine, if we're more open to the Apps Engine, we can bring thousands ideas into life 😄

**Review by d-gubert — APPROVED** [review]

**Pull request merged**